server/api/db/sqldb/db.py

```diff
@@ -4214,6 +4215,42 @@ def create_data_version(self, session, version):
         data_version_record = DataVersion(version=version, created=now)
         self._upsert(session, [data_version_record])
 
+    def store_alert_template(
+        self, session, template: mlrun.common.schemas.AlertTemplate
+    ) -> mlrun.common.schemas.AlertTemplate:
+        template_record = self._get_alert_template_record(
+            session, template.template_name
+        )
+        if not template_record:
+            return self._create_alert_template(session, template)
+        template_record.full_object = template.dict()
+
+        self._upsert(session, [template_record])
+        return self._transform_alert_template_record_to_schema(
+            self._get_alert_template_record(session, template.template_name)
+        )
+
+    def _create_alert_template(
+        self, session, template: mlrun.common.schemas.AlertTemplate
+    ) -> mlrun.common.schemas.AlertTemplate:
+        template_record = self._transform_alert_template_schema_to_record(template)
+        self._upsert(session, [template_record])
+        return self._transform_alert_template_record_to_schema(template_record)
+
+    def delete_alert_template(self, session, name: str):
+        self._delete(session, AlertTemplate, name=name)
+
+    def list_alert_templates(self, session) -> list[mlrun.common.schemas.AlertTemplate]:
+        query = self._query(session, AlertTemplate)
+        return list(map(self._transform_alert_template_record_to_schema, query.all()))
+
+    def get_alert_template(
+        self, session, name: str
+    ) -> mlrun.common.schemas.AlertTemplate:
+        return self._transform_alert_template_record_to_schema(
+            self._get_alert_template_record(session, name)
+        )
```

Comment:
This will throw an ugly error that is not clear to the user in case `_get_alert_template_record` returns `None`. Suggest to separate it - first `_get`, and explode with a meaningful `mlrun.errors.MLRunNotFoundError` error if not found, then package and return.

Reply:
_transform_alert_template_record_to_schema() handles None, so it will return None

Reply:
Yes, I was looking at `_transform_alert_template_schema_to_record` which doesn't do this check. 
Usually we raise already in the db code, but I do see that you check in the endpoint code and raise if `None` is returned, so that works.